### DISCUSSION TITLE
Position of Annotations in relation to Fields?

### DISCUSSION THREAD

**Discussion by Vigrond** [py-pdf/pypdf#1265] [Q&A]
I want to write a bit of text next to a field (not in the field).

For example, to label a checkbox.

I have read the documentation, but it is unclear to me if PyPDF2 can do this.  There does not seem to be positional information related to fields, so it is hard to see where to insert a AnnotationBuilder.free_text

Any hints appreciated.

**Comment by pubpub-zz** [comment]
the example inhere should be a good example
https://pypdf2.readthedocs.io/en/latest/user/adding-pdf-annotations.html

**Reply by MartinThoma** [reply]
I should probably add more information to the rect parameter. I'll do that next week.

**Reply by Vigrond** [reply]
The example got me started, but in order to find the `rect` property on a `field` I needed to search `page['Annots']`

Here is a bad code example for the benefit of anyone

```
# Find button/checkbox and text fields
field_annots = [annot.get_object() for annot in page['/Annots'] if annot.get_object()['/FT'] in ['/Btn', '/Tx']]

# narrow down to the annotation for a specific field
field_name = "example"
annot = next((field_annot for field_annot in field_annots if field_annot['/T'] == field_name), None)

# if field annotation is found, retrieve the rect property and build new text annotation
if annot:
  rect = annot['/Rect']
  annotation = AnnotationBuilder.free_text(
      "example text"
      rect=rect,
      ...
  )

```
